rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,143 @@
+
+The flag `--bake-image=true` will cause the tool to also run `minaform bake-image` to build a new docker image with all of the require network configuration and s3-cached files baked-in.
+
+Two flags are availible to specify which version of the daemon to run, `--daemon-image` and `--daemon-git-sha` where daemon-git-sha calculates the image tag based on the given git tag (Pain Point #5). 
+
+Additional flags can be provided to the create-network command for changing the quantities of fish, whales, seeds, bots, etc. These quantaties are provided at this stage so that the genesis ledger can be populated with proper accounts for these services.
+
+Once the network has been defined and set up, you can always make manual changes to the terraform files to customize the new network, or just proceed to `minaform start <testnet_name>`. 
+
+#### terraform.env
+
+The configured docker image, git tag, deb package tag and the image generated by create-network (with deb package + keys + genesis proof) are stored alongside the default network configuration (prod or qa) in a `terraform.env` file and in `export TF_BLAH=blah` format so that those variables can be easily validated by the deploying engineer, and updated for each iteration of a testnet.  Google cloud storage paths to the keyfiles and ledger are also provided in this file for ease of use and visibility for anyone working with this network, and for use in `minaform get-network`.
+
+Any additional common configuration parameters like log-level can also be added to the terraform environment in this way and exported as terraform environment variables so that those settings can even more easily be changed without touching the terraform. At any time as the default terraform testnets evolve we can use environment variables in more places identified to be commonly variable across networks.
+
+Potentially this environment can be stored in GCP alonside keys and other things, while the coda-automation repo only houses the default network configs but I expect that providing a terraform file for each network even when the tf does not need to be updated will ensure that this solution remains compatible with the good parts of old workflows (terraform's flexibility) without the bad parts (hard to standardize when everything is flexible). If at some point in the future the vast majority of minaform networks have been deployed without edits to the tf then the defaults can be changed to only output a tf file for each network when an explicit flag is provided.
+
+### Minaform get-network
+
+The get-network command is meant to be used alongside create-network to easily allow engineers to collaborate on deployments of a testnet. `minaform get-network <testnet_name>` will use the terraform folder and `terraform.env` to download the keysets and genesis ledger for the given testnet so that you can start, stop, restart, or upgrade a network that you did not create.
```

Comment:
Again some sort of spec (perhaps expressed through *tfvars*) could achieve the same result here. I really see all of the below subcommands replaced by some form of:

e.g .`terraform plan|apply|destroy -var-file={andrew-minanet, ahmad-minanet, mina-qanet, product-staging-net, etc}.tfvars`